src/main/java/com/fasterxml/jackson/databind/jsontype/NamedType.java

```diff
@@ -14,10 +14,10 @@
     protected String _name;
     
     public NamedType(Class<?> c) { this(c, null); }
-    
+
     public NamedType(Class<?> c, String name) {
         _class = c;
-        _hashCode = c.getName().hashCode();
+        _hashCode = (c.getName() + (name != null ? name : "")).hashCode();
```

Comment:
Will also need to change `equals()` to compare name now (to be compatible with semantic definition of `hashCode()`), I think.